x-pack/plugins/infra/public/hooks/use_alerts_history.ts

```diff
@@ -0,0 +1,186 @@
+          featureIds,
+          http,
+          ruleId,
+          signal: abortCtrlRef.current.signal,
+          dateRange,
+        });
+
+      if (error) throw error;
+      if (!isCancelledRef.current) {
+        setTriggeredAlertsHistory((oldState: AlertsHistory) => ({
+          ...oldState,
+          alertsHistory: {
+            totalTriggeredAlerts,
+            histogramTriggeredAlerts,
+            avgTimeToRecoverUS,
+          },
+          isLoadingAlertsHistory: false,
+        }));
+      }
+    } catch (error) {
+      if (!isCancelledRef.current) {
+        if (error.name !== 'AbortError') {
+          setTriggeredAlertsHistory((oldState: AlertsHistory) => ({
+            ...oldState,
+            isLoadingAlertsHistory: false,
+            errorAlertHistory: error,
+            alertsHistory: undefined,
+          }));
+        }
+      }
+    }
+  }, [dateRange, featureIds, http, ruleId]);
+  useEffect(() => {
+    loadRuleAlertsAgg();
+  }, [loadRuleAlertsAgg]);
+
+  return triggeredAlertsHistory;
+}
+
+export async function fetchTriggeredAlertsHistory({
+  featureIds,
+  http,
+  ruleId,
+  signal,
+  dateRange,
+}: {
+  featureIds: ValidFeatureId[];
+  http: HttpSetup;
+  ruleId: string;
+  signal: AbortSignal;
+  dateRange: {
+    from: string;
+    to: string;
+  };
+}): Promise<FetchAlertsHistory> {
+  try {
+    const res = await http.post<AsApiContract<any>>(`${BASE_RAC_ALERTS_API_PATH}/find`, {
+      signal,
+      body: JSON.stringify({
+        size: 0,
+        feature_ids: featureIds,
+        query: {
+          bool: {
+            must: [
+              {
+                term: {
+                  [ALERT_RULE_UUID]: ruleId,
+                },
+              },
+              {
+                range: {
+                  [ALERT_TIME_RANGE]: dateRange,
+                },
+              },
+            ],
+          },
+        },
+        aggs: {
+          histogramTriggeredAlerts: {
+            date_histogram: {
+              field: ALERT_START,
+              fixed_interval: '1d',
+              extended_bounds: {
+                min: dateRange.from,
+                max: dateRange.to,
+              },
+            },
+          },
+          avgTimeToRecoverUS: {
+            filter: {
+              term: {
+                [ALERT_STATUS]: 'recovered',
+              },
+            },
+            aggs: {
+              recoveryTime: {
+                avg: {
+                  field: ALERT_DURATION,
+                },
+              },
+            },
+          },
+        },
+      }),
+    });
+    const totalTriggeredAlerts = res?.hits.total.value;
+    const histogramTriggeredAlerts = res?.aggregations?.histogramTriggeredAlerts.buckets;
+    const avgTimeToRecoverUS = res?.aggregations?.avgTimeToRecoverUS.recoveryTime.value;
+
+    return {
+      totalTriggeredAlerts,
+      histogramTriggeredAlerts,
+      avgTimeToRecoverUS,
+    };
+  } catch (error) {
```

Comment:
I assume this error is an instance of `Error`, but in the interface, we have `error?: string;`, how does it work?